File: projects/order-creation/core/src/main/java/com/robinfood/core/entities/transactionrequestentities/TransactionRequestEntity.java
package com.robinfood.core.entities.transactionrequestentities;

import com.robinfood.core.entities.OrderFiscalIdentifierEntity;
import com.robinfood.core.entities.redeemcoupon.PerformCouponResponseEntity;
import java.math.BigDecimal;
import java.util.List;
import java.util.UUID;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@AllArgsConstructor
@Data
@NoArgsConstructor
public class TransactionRequestEntity {

    private AddressEntity address;

    private Boolean alreadyValidated;

    private CompanyEntity company;

    private List<CouponEntity> coupons;

    private List<PerformCouponResponseEntity> couponResponseEntities;

    private DeliveryEntity delivery;

    private DeviceEntity device;

    private List<DeductionEntity> deductions;

    private Long flowId;

    private Long id;

    private List<OrderRequestEntity> orders;

    private OrderFiscalIdentifierEntity buyer;

    private Boolean paid;

    private List<PaymentMethodEntity> paymentMethods;

    private List<PaymentMethodEntity> paymentsPaid;

    private BigDecimal total;

    private NewUserEntity user;

    private UUID uuid;

    private BigDecimal co2Total;

    /**
     * The constructor is overloaded, because the addresses attribute is optional.
     */
    public TransactionRequestEntity(
            Boolean alreadyValidated,
            CompanyEntity company,
            DeliveryEntity delivery,
            DeviceEntity device,
            Long flowId,
            Long id,
            List<OrderRequestEntity> orders,
            OrderFiscalIdentifierEntity orderFiscalIdentifier,
            Boolean paid, List<PaymentMethodEntity> paymentMethods,
            BigDecimal total, NewUserEntity user) {

        this.alreadyValidated = alreadyValidated;
        this.company = company;
        this.delivery = delivery;
        this.device = device;
        this.flowId = flowId;
        this.id = id;
        this.orders = orders;
        this.buyer = orderFiscalIdentifier;
        this.paid = paid;
        this.paymentMethods = paymentMethods;
        this.total = total;
        this.user = user;
    }
}

File: projects/posv2/core/src/main/kotlin/com/robinfood/core/entities/transactionrequest/TransactionRequestEntity.kt
package com.robinfood.core.entities.transactionrequest

import com.robinfood.core.constants.GlobalConstants.DEFAULT_BOOLEAN_VALUE
import com.robinfood.core.entities.OrderCouponEntity
import java.math.BigDecimal

data class TransactionRequestEntity(
    val buyer: BuyerEntity?,
    val coupons: List<OrderCouponEntity>,
    val company: CompanyEntity,
    val co2Total: BigDecimal,
    val device: DeviceEntity,
    val discounts: List<DiscountEntity>,
    val flowId: Long,
    val id: Long? = null,
    val delivery: IntegrationDataEntity?,
    val orders: List<OrderEntity>,
    val origin: OriginEntity,
    val paid: Boolean = !DEFAULT_BOOLEAN_VALUE,
    val paymentMethods: List<PaymentMethodEntity>,
    val paymentsPaid: List<PaymentMethodEntity>,
    val total: BigDecimal,
    val totalPaidPayments: BigDecimal,
    val updateOrder: Boolean,
    val user: TransactionUserEntity,
    val uuid: String? = null
)
